driver-core/src/test/functional/com/mongodb/client/model/expressions/ExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,646 @@
+import org.bson.codecs.BsonDocumentCodec;
+import org.bson.codecs.BsonValueCodecProvider;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+import org.bson.json.JsonReader;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.client.model.Aggregates.addFields;
+import static com.mongodb.client.model.expressions.Expressions.CURRENT;
+import static com.mongodb.client.model.expressions.Expressions.ofNull;
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+/**
+ * This test file groups expressions of each type under one test method. Each of
+ * these methods begins by showing how to express literals of the relevant type.
+ * The ensuing assertions then show how to express the computation in Java (when
+ * reasonable), then using the API under test, and then in MQL.
+ */
+@SuppressWarnings({
+        // for demonstration:
+        "PointlessBooleanExpression",
+        "PointlessArithmeticExpression",
+        "ConstantConditions",
+        // for clarity:
+        "Convert2MethodRef"})
+public class ExpressionsFunctionalTest extends OperationTest {
+
+    @Test
+    public void booleanExpressionsTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#boolean-expression-operators
+        // (Complete as of 6.0)
+
+        // literals
+        BooleanExpression tru = Expressions.ofTrue();
+        BooleanExpression fal = Expressions.ofFalse();
+        assertExpression(true, tru, "true");
+        assertExpression(false, fal, "false");
+        assertTrue(tru instanceof BooleanExpression);
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/or/
+        assertExpression(true || false, tru.or(fal), "{'$or': [true, false]}");
+        assertExpression(false || true, fal.or(tru), "{'$or': [false, true]}");
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/and/
+        assertExpression(true && false, tru.and(fal), "{'$and': [true, false]}");
+        assertExpression(false && true, fal.and(tru), "{'$and': [false, true]}");
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/not/
+        assertExpression(!true, tru.not(), "{'$not': true}");
+        assertExpression(!false, fal.not(), "{'$not': false}");
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/cond/
```

Comment:
This was not static.

At some point, I considered chaining cond off of the preceding expression, exp.cond(condition, left, right). This would have allowed the use of lambdas in the left and right positions, which made use of whatever exp was (this might have even used $let in the background). But this was too cumbersome - if that was needed, users can go with switchOn.